### PR TITLE
Refactor to remove external dependencies from repo

### PR DESCRIPTION
**Pull request by daniel-ac-martin**
* Refactors the repositories to remove all external dependencies from
  version control and simplify directory structure.
* Eliminates Grunt in favour of NPM.
* Adds documentation to README explaining how to build.
* Automates package generation.
* Removes other extraneous files.
* Renames theme from 'govuk_latest' to just 'govuk'.

### CODE REVIEW

**Comment by daniel-ac-martin** [comment]
I'm not happy with the `package.json` as I think it could be neatened up a lot. But I'm fairly happy with the overall strategy. What do you think?

**Comment by paulwaitehomeoffice** [comment]
Just taking a proper look, but overall this is really helpful, much better.

**Comment by paulwaitehomeoffice** [comment]
This is great stuff: thanks for going through all this, it’s much simpler now.

A couple of minor ideas:

1. Do we want to move `rm -rf govuk/login/resources/` into a `build:clean` script, and add it to the start of `build` (`"build": "npm-run-all build:clean build:assets build:sass"`)? So we start with a clean slate each time?

2. I think I can see the argument for excluding `govuk/login/resources/` from source control.

    However, if a future change to one of the GOV.UK dependencies ends up causing some weirdness with the theme, might it be easier to spot if we’ve got a change history for the built dependencies there in the theme project, rather than having to go to the git repos for the dependencies and work out how their changes would have affected the built output?

    Maybe it would just clutter up our repo, I’m not sure.

3. It’s a separate issue really, but have you had any thoughts about how we’d organise doing different versions of the themes? (Like your social-providers-only version, or my version for internal projects following the Home Office beta header pattern.) Maybe long-lived git branches?

    I only ask because the existing theme currently lives in a top-level directory called govuk. If we were going to have multiple theme directories in this project, maybe it should be themes/govuk.

File: README.md [hunk 1]
```diff
@@ -33,29 +36,31 @@ The theme can be built using the following commands:
...
 ```bash
 $ npm install
 $ npm run build
 $ docker-compose up
 ```
-The Keycloak server will now be available on <http://localhost:8080>.
+The Keycloak server will now be available on <http://localhost:8080>. You can log into the Administration Console using “**admin**” as both username and password.
 
-To enable the theme in the local Keycloak server, log in (using “**admin**” as both username and password), and follow steps 2 and 3 in [the “Using” section above](#using).
+<img alt="" src="docs/images/keycloak-local.png" width="512" height="300">
```
Comment (daniel-ac-martin) [inline README.md]: I think there's markdown code for this.
one does not simply write markdown

Comment (paulwaitehomeoffice) [inline README.md]: 1. Grand, I’m happy to leave as-is.

2. Yeah I was thinking more for the convenience of inspecting changes in GOV.UK stuff between versions if necessary. But you’re right, it’s unlikely to be a big issue.

One thing that might be more fiddly is keeping to date with the original Keycloak theme files as they change between Keycloak versions. Obviously that’ll be fiddly anyway, given that we’re changing them, but as things stand a developer who’s trying to keep our theme up-to-date with the latest version of Keycloak will have to go and download both the latest version of Keycloak, and whatever version our theme was based on, and diff them. That was my thinking when including the original Keycloak theme files in the repo.

3. Yeah a different repo for every theme feels unnecessary at this point — I’m assuming each of our variations are going to be reasonably small changes to the main theme.

If we had different directories for each new theme, would we copy all the theme files into it? I worry that would make the different themes difficult to maintain if and when the theme template files change in future.

**Comment by daniel-ac-martin** [comment]
[Quote from line comment - so it doesn't get lost!]

1. Grand, I’m happy to leave as-is.

2. Yeah I was thinking more for the convenience of inspecting changes in GOV.UK stuff between versions if necessary. But you’re right, it’s unlikely to be a big issue.

    One thing that might be more fiddly is keeping to date with the original Keycloak theme files as they change between Keycloak versions. Obviously that’ll be fiddly anyway, given that we’re changing them, but as things stand a developer who’s trying to keep our theme up-to-date with the latest version of Keycloak will have to go and download both the latest version of Keycloak, and whatever version our theme was based on, and diff them. That was my thinking when including the original Keycloak theme files in the repo.

3. Yeah a different repo for every theme feels unnecessary at this point — I’m assuming each of our variations are going to be reasonably small changes to the main theme.

    If we had different directories for each new theme, would we copy all the theme files into it? I worry that would make the different themes difficult to maintain if and when the theme template files change in future.

**Comment by daniel-ac-martin** [comment]
Regarding point 2. The version of Keycloak is could be stored in the form of the tag on the Keycloak docker image in `docker-compose.yml`. Right now it is `latest` but we should probably change that so it is clear what we have developed against.

**Comment by daniel-ac-martin** [comment]
Regarding point 3. It would appear that we should be able to base new themes on the GovUK theme. I'm guessing that should mean that we only need to include a few of the theme files. (The ones we want to overwrite.)

**Comment by paulwaitehomeoffice** [comment]
> Regarding point 2. The version of Keycloak is could be stored in the form of the tag on the Keycloak docker image in ./docker-compose.yml. Right now it is 'latest' but we should probably change that so it is clear what we have developed against.

Gotcha — yeah nice, that does it I reckon.

> Regarding point 3. It would appear that we should be able to base new themes on the GovUK theme. 

Oh yes of course, that’s how they work isn’t it. Grand, separate directories should work fine then.

**Pull request merged**